### ISSUE TITLE
Query

### ISSUE THREAD

**Issue by SupriyaUpadhyaya** [TaoShi1998/SSLCL#3]
Hello @TaoShi1998,

Thanks for [confirmation](https://github.com/TaoShi1998/SSLCL/issues/2) that we could use either pre-trained or non pre-trained network for 'ERC model'.

Additionally, requesting your to please confirm on below queries:

1. We are training both 'ERC model' and 'Label learning' model using back propagation. The SSLCL + cross entropy loss is calculated for ERC model and same is back propagated through both ERC network and label learning network using two different optimizers in PyTorch (code screenshot attached). Please could you provide feedback if this training approach matches your paper's recommendations.

2. If a non pre-trained ERC model is used then the parameters of the ERC model are to be optimized based on SSLCL + cross entropy loss. The SSLCL loss is dependent on the label embeddings. The label embeddings inturn need to be optimized based on the SSLCL + cross entropy loss from ERC model(which is also not yet optimized). With such interdependency, how is it guaranteed that the optimal label embeddings are learned ?

<img width="1021" alt="sslcl-joint-training" src="https://github.com/TaoShi1998/SSLCL/assets/18483906/9d6f6f50-1e73-4dce-9af7-a42c26fd0e18">

Thanks and regards,
Supriya

**Comment by TaoShi1998** [comment]
Hi @SupriyaUpadhyaya, 

For the first question, it seems correct to me but it really depends on the specific scenario. If the loss works fine (e.g., showing a consistent decrease without any sudden fluctuations or increases), then you should be fine. As for the second question, please refer to https://arxiv.org/pdf/1811.08979.pdf, where you can find the proof.

Best, 
Tao
